server.go

```diff
@@ -3583,12 +3596,89 @@ func (s *server) peerTerminationWatcher(p *peer.Brontide, ready chan struct{}) {
 		}
 
 		srvrLog.Debugf("Attempting to re-establish persistent "+
-			"connection to peer %v", p)
+			"connection to peer %v", p.IdentityKey())
 
-		s.connMgr.Connect(connReq)
+		s.connectToPersistentPeer(pubStr)
 	}()
 }
 
+// connectToPersistentPeer uses all the stored addresses for a peer to attempt
+// to connect to the peer. It creates connection requests if there are
+// currently none for a given address and it removes old connection requests
+// if the associated address is no longer in the latest address list for the
+// peer.
+func (s *server) connectToPersistentPeer(pubKeyStr string) {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	// Exit if this is not a persistent peer.
```

Comment:
Is this necessary given it's checked above?

Reply:
cool, yeah, removed 👍